### PR TITLE
Enable Source-Id as a header field

### PR DESCRIPTION
**Pull request by abbi-gaurav**
**Description**
- Adjust publish API to accept source id field from the header
- Add tests and validations for sourceid from payload and header
- Update Makefile to remove push as it is no longer possible to run
- Move vet and test targets to a common Makefile at the root
- Update Jenkins file accordingly.
- Add stages to run and vet the event-bus source code as first stages
- Ensure source id in the payload is given precedence when it is present in both header and payload
- Did minor clean up to remove some unwanted functions and variables
- Update the OpenAPI definition.
- Remove the source field. This was a miss in the api.yaml

**Related issue(s)**
See also #1048

### CODE REVIEW

File: components/event-bus/cmd/event-bus-publish/handlers/handlers.go [hunk 1]
```diff
@@ -90,6 +90,11 @@ func handlePublishRequest(w http.ResponseWriter, r *http.Request, publisher *con
 		return
 	}
 	publishRequest, err := parseRequest(body)
+
+	if len(publishRequest.SourceID) == 0 {
```
Comment (marcobebway) [inline components/event-bus/cmd/event-bus-publish/handlers/handlers.go]: At this point `err` may not be nil.

Please move this check after calling `ValidatePublish`.

Comment (abbi-gaurav) [reply components/event-bus/cmd/event-bus-publish/handlers/handlers.go]: Thanks for the feedback. As discussed the logic will be added after `err` nil check but before `ValidatePublish` as we need to add the sourceId from the header before validating.

**Comment by abbi-gaurav** [comment]
@marcobebway applied the feedback. Please have a look

File: components/event-bus/Jenkinsfile [hunk 2]
```diff
@@ -50,8 +50,12 @@ podTemplate(label: label) {
                             + "./...")
                         }
 
-                        stage("build and test - event-bus-publish") {
-                            execute("/cmd/event-bus-publish", "make clean build")
+                        stage("test - event-bus") {
```
Comment (marcobebway) [inline components/event-bus/Jenkinsfile]: 👍

**Pull request merged**